extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/InMemoryQueryRepository.java

```diff
@@ -247,4 +259,39 @@ private void updateCache() {
             }
         }
     }
+
+    @Override
+    protected void runOneIteration() throws Exception {
+        lock.lock();
+        try {
+            updateCache();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    protected Scheduler scheduler() {
+        return Scheduler.newFixedRateSchedule(0L, 1, TimeUnit.MINUTES);
```

Comment:
Lets pass the scheduler into this class' constructor.

Reply:
done